packages/fhir-sanitize-param/index.js

```diff
@@ -72,142 +72,230 @@ function findMatchForConfig(name, params) {
+	invariant(typeof value === type, mismatchError(type, name));
+	return value;
+}
 
-	return {
-		dateString,
-		modifier,
-		date,
-	};
+/**
+ * @function sanitizeDate
+ * @description Sanitize date values
+ * @param name
+ * @param inputValue
+ * @param type
+ * @returns {*}
+ */
+function sanitizeDate(name, inputValue, type = 'date') {
+	let { prefix, value } = splitPrefixFromValue(inputValue);
+	invariant(moment(value).isValid(), mismatchError(type, name));
+	return prefix + value;
 }
 
 /**
- * @function coerceValue
- * @throws
- * @description Given a value and a type, attempt to coerce the value to the
- * correct type
- * @param {ParamConfig} config
- * @param {*} value - Value that will be coerced
- * @return {*}
+ * @function sanitizeNumber
+ * @description Sanitize number values
+ * @param name
+ * @param inputValue
+ * @param type
+ * @returns {*}
  */
-function coerceValue(config, value) {
-	let { name, type } = config;
-	let result;
-	// Check all expected types, the config may need to be updated to reflect
-	// FHIR specific types
-	switch (type) {
-		case 'number':
-		case 'integer':
-		case 'decimal':
-			// This validator only accepts strings
-			result = validator.toFloat('' + value);
-			// Throw on invalid results
-			invariant(
-				typeof result === 'number' && !Number.isNaN(result),
-				mismatchError(type, name),
-			);
-			break;
-		case 'date':
-			// Dates will have a custom format as well
-			// for the values since they can have modifiers
-			// in a different format, like gte
-			result = parseAndSanitizeDate(value);
-			// Throw if unable to make a moment date from value
-			invariant(
-				moment(result.dateString).isValid() && moment.isMoment(result.date),
-				mismatchError(type, name),
-			);
-			break;
-		case 'boolean':
-			result = validator.toBoolean(value, true);
-			// Throw if we dont have a boolean value
-			invariant(typeof result === 'boolean', mismatchError(type, name));
-			break;
-		case 'uri':
-		case 'url':
-		case 'string':
-		case 'reference':
-			// strip any html tags from the query
-			// xss helps prevent html from slipping in
-			// strip a certain range of unicode characters
-			// replace any non word characters
-			result = validator.stripLow(xss(sanitize(value)));
-			// Throw if this somehow passes above
-			invariant(typeof result === 'string', mismatchError(type, name));
-			break;
-		case 'token':
-			// Throw if the value is not a string, because we will not be able
-			// to make a token out of it
-			invariant(typeof value === 'string', mismatchError(type, name));
-			// tokens may contain codes and systems that are separated by pipes
-			// make sure to correctly parse these values
-			result = parseAndSanitizeToken(value);
+function sanitizeNumber(name, inputValue, type = 'number') {
+	let { prefix, value } = splitPrefixFromValue(inputValue);
+	const coercedVal = validator.toFloat('' + value);
+	invariant(
+		typeof coercedVal === 'number' && !Number.isNaN(coercedVal),
```

Comment:
I think you're looking for the global isNaN and not the Number.isNaN.  There's a big difference on detection.  Number.isNaN doesn't do what it sounds like.

Reply:
could use the global isNaN() to condense this functionality